hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AFileSystem.java

```diff
@@ -241,26 +242,17 @@ public StorageStatistics provide() {
                     }
                   });
 
-      int maxThreads = conf.getInt(MAX_THREADS, DEFAULT_MAX_THREADS);
-      if (maxThreads < 2) {
-        LOG.warn(MAX_THREADS + " must be at least 2: forcing to 2.");
-        maxThreads = 2;
-      }
+      int maxThreads = getMaxThreads(conf, MAX_THREADS, DEFAULT_MAX_THREADS);
```

Comment:
I'm assuming the checks are here for a reason. Unless lazy xfer manager does the uprate, it'll need reinstatement. Doing it in the manager would be best